### ISSUE TITLE
[Bug]: Synchornization is unreliable when sharing a group folder to a Team (circles)

### ISSUE THREAD

**Issue by mgallien** [nextcloud/server#49126]
### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [x] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Changes are not visible by the desktop files client when syncing as an user receiving a group folders shared via a Team (i.e. circles) in view-only mode.
Etag are not propagated on changes made by the sharing user to the group folder.

### Steps to reproduce

1. Create a group folder `test`
2. Create an user group `group` and assign `user1` to it
3. Assign full access to `test` group folder to user group `group`
4. Create a Team called `viewTeam` as `userA` and add `userB` into it
5. As `userA`, share `group` folder to \viewTeam` in view only mode
6. Setup desktop files client for user `userB` and sync
7. As `userA` do changes to group folder `group`
8. As `userB` sync them
9. As `userB` see the changes in web UI

### Expected behavior

The expected behavior would be that `userB` immediately sees the updated state of any items related to `group` group folder.
One can see the bug in web UI as the modification time of teh top folder of `group` group folder is visibly not updated in web UI when refreshing the page after `userA` did changes.

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by icewind1991** [comment]
Interestingly, `occ info:file ...` does show the correct, updated, etag. But dav shows a different one...

**Comment by icewind1991** [comment]
Something seems to be caching the metadata, setting the configured caches to `NullCache` resolves the issue.

**Comment by icewind1991** [comment]
https://github.com/nextcloud/circles/pull/972 added the caching for circle shares

**Comment by ArtificialOwl** [comment]
yes, but the `save()` method is called and reset the cache when generating a new share on a groupfolder

**Comment by icewind1991** [comment]
The issue isn't when creating a new share, it's when editing files inside the share. The share recipient will see the old metadata for the cache duration

**Comment by ArtificialOwl** [comment]
Oh. got it

**Comment by jospoortvliet** [comment]
So this should be fixed with https://github.com/nextcloud/circles/pull/1745

**Comment by ornanovitch** [comment]
The fix was reverted (https://github.com/nextcloud/circles/pull/1830)

See this new very similar issue: https://github.com/nextcloud/server/issues/57973
